src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/UserEntity.kt

```diff
@@ -33,6 +33,20 @@ interface UserRepository : JpaRepository<UserEntity, UUID>, JpaSpecificationExec
     nativeQuery = true
   )
   fun findQualifiedAssessorWithLeastPendingAllocations(requiredQualifications: List<String>, totalRequiredQualifications: Long): UserEntity?
+
+  @Query(
+    """
+    SELECT u.*, ura.* 
+    FROM "users"  u
+	    LEFT JOIN user_role_assignments ura ON ura.user_id = u.id 
+    WHERE ura.role = 'MATCHER'
+    ORDER BY 
+      (SELECT COUNT(1) FROM placement_requests a WHERE a.allocated_to_user_id = u.id AND a.booking_id IS NULL) ASC 
+    LIMIT 1
+    """,
+    nativeQuery = true
+  )
+  fun findMatcherWithLeastPendingPlacementRequests(): UserEntity?
```

Comment:
Would random allocation be a fairer strategy, in that it wouldn't over-allocate to those matchers who are doing their work quickest?